executor/cook/executor.py

```diff
@@ -0,0 +1,384 @@
+        :return: True if the process was killed, False if it terminated naturally.
+    """
+    process, _, _ = process_info
+
+    process_killed = False
+    while is_running(process):
+
+        if stop_signal.isSet():
+            logging.info('Executor has been instructed to terminate')
+            kill_task(driver, task, process_info)
+            process_killed = True
+            break
+
+        time.sleep(1)
+    return process_killed
+
+
+def manage_task(driver, task, stop_signal, completed_signal, max_message_length, progress_sample_interval_ms,
+                sandbox_location, stdout_name, stderr_name, progress_output_name, progress_regex_string):
+    """
+        Manages the execution of a task.
+        :return: Nothing
+    """
+    try:
+        task_id = get_task_id(task)
+        sandbox_message = json.dumps({'sandbox-location': sandbox_location, 'task-id': task_id})
+        send_message(driver, sandbox_message, max_message_length)
+
+        process_info = launch_task(driver, task, stdout_name, stderr_name)
+        process, _, _ = process_info
+
+        progress_watcher = ProgressWatcher(progress_output_name, progress_regex_string, completed_signal)
+        progress_updater = ProgressUpdater(progress_sample_interval_ms)
+        progress_complete_event = launch_progress_tracker(driver, task, max_message_length, progress_watcher,
+                                                          progress_updater)
+
+        process_killed = await_process_completion(driver, task, stop_signal, process_info)
+        exit_code = process.returncode
+
+        # set the completed signal to trigger progress updater termination and await progress
+        # updater termination if executor is terminating normally
+        completed_signal.set()
+        if not stop_signal.isSet():
+            logging.info('Awaiting progress updater completion with timeout of one second')
+            progress_complete_event.wait(1)
+
+        logging.info('Process completed with exit code {}'.format(exit_code))
+        exit_message = json.dumps({'exit-code': exit_code, 'task-id': task_id})
+        send_message(driver, exit_message, max_message_length)
+
+        # send the latest progress state if available
+        latest_progress = progress_watcher.current_progress()
+        progress_updater.send_progress_update(driver, task_id, max_message_length, latest_progress, force_send=True)
+
+        if not process_killed:
+            task_state = 'TASK_FINISHED' if exit_code == 0 else 'TASK_FAILED'
+            update_status(driver, task, task_state)
+
+    except Exception:
+        logging.exception('Error in executing task')
+        update_status(driver, task, 'TASK_FAILED')
+
+    finally:
+        # ensure completed_signal is set
+        completed_signal.set()
+
+
+def run_mesos_driver(stop_signal, max_message_length, progress_sample_interval_ms, sandbox_location, progress_output_name,
```

Comment:
The argument list is long.  Is that pythonic?  Is there a more idiomatic way to pass this config data around?